src/gettext/check.rs

```diff
@@ -0,0 +1,75 @@
+const fn in_set(pat: &[u8], range: std::ops::Range<usize>, keys: &[&str]) -> bool {
+    let mut k = 0;
+    while k < keys.len() {
+        'keycheck: {
+            if range.end - range.start == keys[k].len() {
+                let mut i = range.start;
+                while i < range.end {
+                    if pat[i] != keys[k].as_bytes()[i - range.start] {
+                        break 'keycheck;
+                    }
+                    i += 1
+                }
+
+                return true;
+            }
+        }
+        k += 1
+    }
+
+    false
+}
+
+pub const fn check_keys(pat: &str, keys: &[&str]) {
+    let pat = pat.as_bytes();
+    let mut i = 0;
+    'outer: while i < pat.len() {
+        if pat[i] == b'{' {
+            let mut j = i + 1;
+            while j < pat.len() {
+                if pat[j] == b'}' {
+                    assert!(
+                        in_set(pat, i + 1..j, keys),
```

Comment:
Bit unfortunate that `&pat[i+1..j]` still isn't stable.

Reply:
And some kind of for loops. Const-Rust feels a bit like a Turing tarpit tbh.